### PR TITLE
CI/CD

### PR DESCRIPTION
**Pull request by stebl**
## Summary

Adds CircleCI config 

## Motivation

CI/CD is great, we want to add it to our auth0-assets

## Details

Relies on CircleCI contexts to inject the correct environment variables,
AUTH0_TENANT_NAME: This tells the script which config file to open
AUTH0_CLIENT_SECRET: Secret key for permissions

Further details added to the README.

I intentionally left the test env out so @claudio-seasoned doesn't have to rotate keys yet in BF development.  It is easy to reconfigure later.  

A small wrapper script calls the auth0 deploy tools.  This was needed to access the client secret from an env var as the CLI will only read it from a file (it will not accept it as a CLI param!)

## Todo

* [ ] Test on CircleCI.  Easiest way to do this is merge to master and see what fails.  The dev/stage/prod envs aren't being used so there's no risk of user impact.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+              if [[ "$audit" != *"ENOAUDIT"* ]]; then
+                break
+              fi
+            done
+
+  deploy:
+    executor: node
+    steps:
+      - with-workspace:
+          command: ./bin/auth0 deploy $AUTH0_TENANT_NAME $AUTH0_CLIENT_SECRET
```
Comment (stebl) [inline .circleci/config.yml:72]: @aldiazok I ended up adding `$AUTH0_TENANT_NAME` to the contexts, is that something that needs to be reflected in the API repo?

File: bin/auth0.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+export type Env = {
+  dumpFromAuth0: typeof dump
+  deployToAuth0: typeof deploy
+}
+
+if (require.main === module) {
+  const command = process.argv[2]
+  const tenant = process.argv[3]
+  const secret = process.argv[4] ?? ''
```
Comment (stebl) [inline bin/auth0.ts:42]: This little snippet is the only untested/non-typesafe code in the script.  

Would it be worthwhile to export a function that accepts `process` and returns these params?  I initially got hung up on this b/c I wasn't sure what the type of `process` is.

Comment (claudio-seasoned) [reply bin/auth0.ts:42]: No idea, but looks like you could just run it locally and test locally to find out, couldn't you?

Comment (stebl) [reply bin/auth0.ts:42]: Yep, works locally

**Pull request merged**